vault/core.go

```diff
@@ -1788,17 +1787,17 @@ func (c *Core) sealInternalWithOptions(grabStateLock, keepHALock, shutdownRaft b
 
 	// If the storage backend needs to be sealed
 	if shutdownRaft {
-		if raftStorage, ok := c.underlyingPhysical.(*raft.RaftBackend); ok {
-			if err := raftStorage.TeardownCluster(c.getClusterListener()); err != nil {
+		if raftBackend := c.getRaftBackend(); raftBackend != nil {
+			if err := raftBackend.TeardownCluster(c.getClusterListener()); err != nil {
 				c.logger.Error("error stopping storage cluster", "error", err)
 				return err
 			}
 		}
-
-		// Stop the cluster listener
-		c.stopClusterListener()
 	}
 
+	// Stop the cluster listener
+	c.stopClusterListener()
```

Comment:
> If we shutdown the cluster listener without shutting down raft then raft cannot operate.
Previously, cluster listeners would only stop on seal if we're dealing with raft. This change makes it so that we are always stopping the cluster listener regardless of whether raft is involved. We always shut down raft first if required and then the cluster listener.